### PR TITLE
ta view, assign instructor,  testing

### PR DESCRIPTION
**Pull request by rachaelcharles**

### CODE REVIEW

**Comment by cooperaustinj** [comment]
Can you run this command in the root directory of the project?

`git rm -f .idea/workspace.xml`

And add this line to the .gitignore
`.idea/workspace.xml`

Then push those changes.  That should remove that file from source control.  I thought I added that file to the gitignore already but I must have not pushed it or something.  That file is different for each person's computer, so we don't want it in source control.

File: TAScheduler/domain/CommandHandler.py [hunk 1]
```diff
@@ -112,7 +136,7 @@ def _NotifyHandler(self, cmd: [str]):
     def _CreateUserHandler(self, cmd: [str]):
         if not self.currentUser or not self.currentUser.RoleIn(Role.Administrator, Role.Supervisor):
             return 'Must be logged in as an Administrator or Supervisor'
-        if len(cmd) != 6 or len(cmd) != 8:
+        if len(cmd) != 6 and len(cmd) != 8:
```
Comment (dakotamaker) [inline TAScheduler/domain/CommandHandler.py]: Good catch here

File: TAScheduler/domain/CommandHandler.py [hunk 2]
```diff
@@ -19,7 +19,7 @@ def ProcessCommand(self, cmdString: str) -> str:
             return 'Badly formatted command'
 
         handler = {
-            'exit': self._ExitHandler,
```
Comment (dakotamaker) [inline TAScheduler/domain/CommandHandler.py]: I am fine with this, but does anyone else have any issues with getting rid of the exit command?

File: TAScheduler/domain/CommandHandler.py [hunk 3]
```diff
@@ -203,7 +246,22 @@ def _ListUsersHandler(self, cmd: [str]):
             return 'Must be logged in as an Administrator or a Supervisor'
 
     def _ListTAsHandler(self, cmd: [str]):
-        return 'List TAs:' + cmd
+        if not self.currentUser:
+            return 'Must be logged in to access TA list.'
+        a = Account.objects.filter(role_id=Role.TA)
+        s = ""
```
Comment (dakotamaker) [inline TAScheduler/domain/CommandHandler.py]: Not the biggest issue, but could probably use more specific variable names

File: TAScheduler/test/test_assignCourseInstructor.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+
+    def test_without_login__assignment_fails(self):
+        ch = CommandHandler()
+        msg = ch.ProcessCommand("assign course instructor Algorithms instructor@email.com")
+        self.assertEqual(msg, 'Must be logged in as an Administrator or Supervisor')
+
+    def test_supervisor_assigns_valid_instructor_and_valid_class__then_assign_succeeds(self):
+        ch = CommandHandler()
+        ch.ProcessCommand("login super@email.com 4444")
```
Comment (dakotamaker) [inline TAScheduler/test/test_assignCourseInstructor.py]: In the setup we should do the login. We can have different tests classes for specific roles. For example, we could do SupervisorEditTestCase, AdminEditTestCase, etc. Just helps the redundancy

File: TAScheduler/test/tests_createCourses.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        ch = CommandHandler()
+        msg = ch.ProcessCommand("create course 'Algorithms'")
+        self.assertEqual(msg, 'Must be logged in as an Administrator or Supervisor')
+        c = Course.objects.filter(course_name='Algorithms')
+        self.assertEquals(c.count(), 0)
+
+
+    def test_with_supervisor_adding_new_course__then_course_created(self):
+        ch = CommandHandler()
+        ch.ProcessCommand("login super@email.com 4444")
```
Comment (dakotamaker) [inline TAScheduler/test/tests_createCourses.py]: Same comment as before with the login stuff

File: TAScheduler/test/test_assignCourseInstructor.py (hunk 4 above)
Comment (cooperaustinj) [reply TAScheduler/test/test_assignCourseInstructor.py]: Good idea.  we can probably wait until next sprint to clean  that up.  Future tests can follow the better pattern though.

Comment (cooperaustinj) [reply TAScheduler/test/test_assignCourseInstructor.py]: Unless Rachel is really ambitious!

**Comment by dakotamaker** [comment]
+ @dakotamaker clean up comments from this PR when doing stories tonight

**Pull request merged**